Modules/TPC/src/CheckOfPads.cxx

```diff
@@ -0,0 +1,337 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   CheckOfPads.cxx
+/// \author Laura Serksnyte, Maximilian Horst
+///
+
+#include "TPC/CheckOfPads.h"
+#include "QualityControl/MonitorObject.h"
+#include "QualityControl/Quality.h"
+#include "QualityControl/QcInfoLogger.h"
+#include "Common/Utils.h"
+
+// ROOT
+#include <TCanvas.h>
+#include <TH1.h>
+#include <TH2.h>
+#include <TPad.h>
+#include <TList.h>
+#include <TPaveText.h>
+
+#include <iostream>
+
+namespace o2::quality_control_modules::tpc
+{
+void CheckOfPads::configure()
+{
+  mMediumQualityLimit = common::getFromConfig<float>(mCustomParameters, "mediumQualityPercentageOfWorkingPads", 0.7);
+  mBadQualityLimit = common::getFromConfig<float>(mCustomParameters, "badQualityPercentageOfWorkingPads", 0.3);
+  std::string CheckChoiceString = common::getFromConfig<std::string>(mCustomParameters, "CheckChoice", "Mean");
+
+  if (size_t finder = CheckChoiceString.find("ExpectedValue"); finder != std::string::npos) {
+    mExpectedValueCheck = true;
+    mFoundCheck = true;
+  }
+  if (size_t finder = CheckChoiceString.find("Mean"); finder != std::string::npos) {
+    mMeanCheck = true;
+    mFoundCheck = true;
+  }
+  if (size_t finder = CheckChoiceString.find("Empty"); finder != std::string::npos) {
+    mEmptyCheck = true;
+    mFoundCheck = true;
+  }
+
+  if (!mFoundCheck) {
+    mMeanCheck = true;
+    ILOG(Warning, Support) << "This Check requires a CheckChoice. There was no value or the given value is wrong or not readable. Chose between 'ExpectedValue'(Compare the pad mean to an expectedValue), 'Mean' (compare pad mean to global mean) and/or 'Empty' (check for a threshold of empty pads). As a default 'Mean' was selected." << ENDM;
+  }
```

Comment:
Why not
```suggestion
  bool foundCheck = false;
  if (checkChoiceString == CheckChoiceExpectedValue) {
    mExpectedValueCheck = true;
    foundCheck = true;
  }
  if (checkChoiceString == CheckChoiceMean) {
    mMeanCheck = true;
    foundCheck = true;
  }
  if (checkChoiceString == CheckChoiceEmpty) {
    mEmptyCheck = true;
    foundCheck = true;
  }
  
    if (!foundCheck) {
    mMeanCheck = true;
    ILOG(Warning, Support) << fmt::format("This Check requires a CheckChoice. There was no value or the given value is wrong or not readable. Chose between '{}'(Compare the pad mean to an expectedValue), '{}' (compare pad mean to global mean) and/or '{}' (check for a threshold of empty pads). As a default '{}' was selected.", CheckChoiceExpectedValue, CheckChoiceMean, CheckChoiceEmpty, CheckChoiceMean) << ENDM;
  }
```

and remove `mFoundCheck`.

Reply:
what about the simplification using direct string comparison?

Reply:
this would not work since we want to be able to run multiple checks at the same time (e.g. empty pad check and comparison against an expected value). So the checkChoiceString could look something like "ExpectedValue,Empty" so a direct string comparison would not work.

Reply:
Ahh, thanks for the explanation! That makes sense.